toolchain/lower/file_context.cpp

```diff
@@ -766,9 +767,82 @@ auto FileContext::BuildFunctionBody(SemIR::FunctionId function_id,
 
   // Emit fingerprint accumulated inside the function context.
   function_lowering.EmitFinalFingerprint();
+  context().di_builder().finalizeSubprogram(subprogram);
+}
+
+auto FileContext::BuildDISubroutineType(
+    const SemIR::Function& function, SemIR::SpecificId specific_id,
+    llvm::SmallVectorImpl<llvm::DIType*>& debug_parameter_types) const
+    -> llvm::DISubroutineType* {
+  auto implicit_param_patterns =
+      sem_ir().inst_blocks().GetOrEmpty(function.implicit_param_patterns_id);
+  auto param_patterns =
+      sem_ir().inst_blocks().GetOrEmpty(function.param_patterns_id);
+
+  auto* void_pointer_debug_type =
+      context().di_builder().createPointerType(nullptr, 8);
+
+  auto get_debug_type = [&](SemIR::TypeId type_id) -> llvm::DIType* {
+    CARBON_CHECK(type_id.has_value());
+    if (!type_id.has_value()) {
```

Comment:
These lines seem to contradict each other

Reply:
Removed the condition in 1059ff48bfbcd75ef5036434f532ee1547904622